cargo-quickinstall/src/command_ext.rs

```diff
@@ -25,17 +25,43 @@ impl CommandExt for Command {
             .and_then(|output| check_status(self, output))
     }
 
-    fn spawn_with_cmd(mut self) -> Result<ChildWithCmd, InstallError> {
+    fn spawn_with_cmd(mut self) -> Result<ChildWithCommand, InstallError> {
         self.spawn()
             .map_err(InstallError::from)
-            .map(move |child| ChildWithCmd { child, cmd: self })
+            .map(move |child| ChildWithCommand { child, cmd: self })
     }
 }
 
 pub struct CommandFormattable<'a>(&'a Command);
 
+fn needs_escape(s: &str) -> bool {
+    s.contains(|ch| !matches!(ch, 'a'..='z' | 'A'..='Z' | '0'..='9' | '-' | '_' | '=' | '/' | ',' | '.' | '+'))
+}
+
 fn write_os_str(f: &mut fmt::Formatter<'_>, os_str: &OsStr) -> fmt::Result {
```

Comment:
we're no longer just writing the string, so write_os_str might not be the best name anymore

```suggestion
fn write_shell_arg_escaped(f: &mut fmt::Formatter<'_>, os_str: &OsStr) -> fmt::Result {
```

Reply:
Done